core/trino-main/src/main/java/io/trino/server/security/HeaderBasedAuthenticator.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.security;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import io.trino.spi.security.AccessDeniedException;
+import io.trino.spi.security.Identity;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+
+import java.security.Principal;
+import java.util.List;
+import java.util.Locale;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.server.security.UserMapping.createUserMapping;
+import static java.util.Objects.requireNonNull;
+
+public class HeaderBasedAuthenticator
+        implements Authenticator
+{
+    private final HeaderBasedAuthenticatorManager authenticatorManager;
+    private final List<String> headersToExtract;
+    private final boolean treatHeadersCaseSensitive;
+    private final UserMapping userMapping;
+
+    @Inject
+    public HeaderBasedAuthenticator(HeaderBasedAuthenticatorConfig authenticatorConfig,
+                                    HeaderBasedAuthenticatorManager authenticatorManager)
```

Comment:
No need to wrap lines. If you do so, please put each argument into separate line. 

```
    public HeaderBasedAuthenticator(
         HeaderBasedAuthenticatorConfig authenticatorConfig,                            
         HeaderBasedAuthenticatorManager authenticatorManager)
```